lib/worker/tests/integration.rs

```diff
@@ -128,6 +139,39 @@ async fn create_publication_and_list_works() {
 
             assert_eq!(format!("{}.{}", ns, rel), p);
 
+            // insert a job and request the deal
```

Comment:
Enhances the existing test to list the deal, it forces an insert into the database and list the deals. this test calls web3 storage, it will fail if we remove the `bafybeibw2zctx4ca3udcfcsizjmo57bomhb6vvzf63rvc25d6hzotncn2i` file from there or if the request fails